yaksh/models.py

```diff
@@ -998,6 +1016,17 @@ def get_user_dir(self):
             os.chmod(user_dir, stat.S_IRWXU | stat.S_IRWXG | stat.S_IRWXO)
         return user_dir
 
+    def _toggle_moderator_group(self, group_name):
+        group = Group.objects.get(name=group_name)
+        if self.is_moderator:
+            self.user.groups.add(group)
+        else:
+            self.user.groups.remove(group)
+
+    def save(self, *args, **kwargs):
```

Comment:
```save``` is overridden here!
And toggle is called for every save!

In case the moderator switches he is no longer in the group, now he is a student, first he edits his profile then save is called with toogle and the student will become moderator again!

Reply:
Good catch, fixed.